compiler/src/dmd/traits.d

```diff
@@ -303,10 +309,18 @@ private Expression pointerBitmap(TraitsExp e, ErrorSink eSink)
     }
 
     Array!(ulong) data;
-    const ulong sz = getTypePointerBitmap(e.loc, t, data, eSink);
+    ulong count;
+    const ulong sz = getTypePointerBitmap(e.loc, t, data, eSink, count);
     if (sz == ulong.max)
         return ErrorExp.get();
 
+    if (count == 0)
```

Comment:
Checking `t.hasPointers()` instead would allow moving this section before the call to `getTypePointerBitmap` and avoid any counting.